setup.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python2.7
+
+# Copyright 2019 Extreme Networks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+
+from setuptools import setup, find_packages
+
+
+PKG_ROOT_DIR = os.path.dirname(os.path.realpath(__file__))
+PKG_REQ_FILE = '%s/requirements.txt' % PKG_ROOT_DIR
+os.chdir(PKG_ROOT_DIR)
+
+
+def get_version_string():
+    version = None
+    sys.path.insert(0, PKG_ROOT_DIR)
+    from st2flake8 import __version__
+    version = __version__
+    sys.path.pop(0)
+    return version
+
+
+def get_requirements():
+    with open(PKG_REQ_FILE) as f:
+        required = f.read().splitlines()
+
+    # Ignore comments in the requirements file
+    required = [line for line in required if not line.startswith('#')]
+    return required
+
+
+setup(
+    name='st2-flake8',
```

Comment:
Should we call it st2-flake8-plugins and ``st2_flake8_plugins`` for package name?

Reply:
My plan is to include other flake8 requirements here.  For example, I will add the flake8-copyright dependency here.  So that when we install this st2-flake8 lib, it'll include other requirements as well.  That's why I didn't call it st2-flake8-plugins because it won't be strictly plugins.

Reply:
I see.

I would actually prefer  us to keep it separate so we can only use license plugin in our open source repos for example.

Reply:
Isn't the flake8 config files what makes flexibility?  On the other hand, I don't agree with the lint-configs repo because it don't give flexibility to individual projects.  This repo on the other hand just package up all the required flake8 dependencies for the config to enable/disable.

Reply:
@Kami ^^^

Reply:
So to clarify, this repo will consist of custom flake8 plugins we write.  Also will include other 3rd party flake8 plugins in requirements.txt.  Unlike hacking, I am not enabling any of these plugins.  All of our st2 repos will just reference this library in respective test-requirements.txt.  Individual st2 repos will have the flexibility with their own flake8 config files to enable/disable the use of these plugins.  This also minimize confusion what 3rd party flake8 plugins are missing and such  because we already defined them in requirements here.  That's my intent.

Reply:
Yeah it depends on the plugins though - if not explicitly disabled by default in code, flake8 plugins are enabled by default (that's the case for most of the hacking plugins).

I'm fine with that approach as long as those flake8 plugins are opt-in aka disabled by default.

Reply:
This is a good point.  I didn't think of it.  Again, I think the flake8 configs have flexibility to disable if that's the case.  I'm not dictating what those flake8 configs will look like here.  Anyway, I think `st2-flake8` is fine as a package name, nice and short.  We know what this is used for and I can update the README.

Reply:
@Kami I updated README.  Are you satisfied yet?  Or must I rename to `st2-flake8-plugins` here before we can move this forward?

Reply:
I would personally still prefer to call it ``st2-flake8-plugins``, but it's not a blocker and I'm also fine with the current name.